cloudfour/cloudfour.com-patterns
# Cleanup gulp tasks

## Overview

Cleans up gulp tasks based on discussions in #677

- JS, Sass, and Types tasks are separated into different files, and can be run individually.
- Adds clean task
- Makes several changes that will make these files eventually more ESM-friendly
  - Removes require-dir, this is not doable with static modules in ESM
  - Changes exports to all be camel case so that they will be able to be exported in ESM. This changes the names of the gulp tasks as well

## Testing

- Make sure all pre-processing related npm tasks can still be run the same way
- Create extra files in `dist` and run `npm run build` and make sure they get removed
- Make sure that running each of the individual build tasks works via `npx gulp buildSass`, etc.
